pkg/controller/common/defaults/pod_template.go

```diff
@@ -55,23 +55,27 @@ func NewPodTemplateBuilder(base corev1.PodTemplateSpec, containerName string) *P
 	return builder.setDefaults()
 }
 
+// getContainer retrieves the existing Container from the pod template
+func (b *PodTemplateBuilder) getContainer() *corev1.Container {
+	for i, c := range b.PodTemplate.Spec.Containers {
+		if c.Name == b.containerName {
+			return &b.PodTemplate.Spec.Containers[i]
+		}
+	}
+	return nil
+}
```

Comment:
can we unit test that to catch future regressions? (it involves pointers and a for loop so a unit test may not hurt 🤷)